File: Silver III/14501/yeolyi.py
N = int(input())
arr = []
for i in range(N):
    arr.append(list(map(int, input().split(" "))))
ret = [0 for _ in range(N + 1)]

for i in reversed(range(N)):
    days = arr[i][0]
    cost = arr[i][1]
    available = i + days <= N
    if available:
        ret[i] = max(ret[i + 1], cost + ret[i + days])
    else:
        ret[i] = ret[i + 1]

print(ret[0])

File: Silver II/2716/yeolyi.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

[...Array(+input[0]).keys()].forEach(n => {
    const str = input[n + 1];
    let cnt = 0;
    let ret = 0;
    for (const c of str) {
        cnt += c === '[' ? 1 : -1
        ret = Math.max(ret, cnt);
    }
    console.log(Math.pow(2, ret));
});
